ccxt_backtesting_exchange/backtester.py

```diff
@@ -86,5 +240,199 @@ def withdraw(self, code: str, amount: float, params={}):
     def fetch_balance(self, params={}):
         """
         Fetch the balance of the backtesting exchange.
+
+        :return: A dictionary of balances indexed by asset.
         """
         return self._balances.set_index("asset").to_dict(orient="index")
+
+    def create_order(
+        self,
+        symbol: str,
+        type: OrderType,
+        side: OrderSide,
+        amount: float,
+        price: float,
+        params={},
+    ) -> any:
+        """
+        Creates a new order and adds it to the order book.
+
+        :param symbol: The trading pair (e.g., "BTC/USDT").
+        :param type: The type of order (e.g., "limit", "market").
+        :param side: The order side ("buy" or "sell").
+        :param amount: The amount of the base asset to trade.
+        :param price: The price at which to place the order.
+        :raises BadSymbol: If the trading pair is invalid.
+        :raises BadRequest: If the order type, side, amount or price is invalid.
+
+        """
+        # Validate inputs
+        if not isinstance(symbol, str) or "/" not in symbol:
+            raise BadSymbol(
+                "Invalid symbol format. Expected 'BASE/QUOTE' (e.g., 'BTC/USDT')."
+            )
+
+        if type not in {"limit", "market"}:
+            raise BadRequest("Invalid order type. Expected 'limit' or 'market'.")
+
+        if side not in {"buy", "sell"}:
+            raise BadRequest("Invalid side. Expected 'buy' or 'sell'.")
+
+        if not isinstance(amount, (int, float)) or amount <= 0:
+            raise BadRequest("Invalid amount. Must be a positive number.")
+
+        if not isinstance(price, (int, float)) or price <= 0:
+            raise BadRequest("Invalid price. Must be a positive number.")
+
+        # Calculate fee
+        fee_cost = amount * price * self._fee
+        base_asset, quote_asset = symbol.split("/")
+        fee = {
+            "currency": quote_asset,
+            "cost": fee_cost,
+            "rate": self._fee,
+        }
+
+        # Update pending balance
+        if side == "buy":
+            trade_value = amount * price + fee_cost  # Buyer needs enough quote currency
+            if self._get_asset_balance(quote_asset, "free") < trade_value:
+                raise InsufficientFunds(
+                    f"Insufficient balance: {quote_asset} balance too low."
+                )
+            self._update_asset_balance(quote_asset, "used", +trade_value)
+            self._update_asset_balance(quote_asset, "free", -trade_value)
+
+        else:  # side == "sell"
+            if self._get_asset_balance(base_asset, "free") < amount:
+                raise InsufficientFunds(
+                    f"Insufficient balance: {base_asset} balance too low."
+                )
+            self._update_asset_balance(base_asset, "used", amount)
+            self._update_asset_balance(base_asset, "free", -amount)
+
+        order_id = len(self._orders)
```

Comment:
I think it is best for us to use UUID for the id, because with what we have here, If I created some order and deleted some then recreate, we can have clashing order ids.

Reply:
Ignore this, I think we need not worry about this, because there is no interface to delete orders directly, except someone hacks the library.

Reply:
exactly.